src/core/default_appearance.js

```diff
@@ -87,6 +87,75 @@ function parseDefaultAppearance(str) {
   return new DefaultAppearanceEvaluator(str).parse();
 }
 
+class AppearanceStreamEvaluator extends EvaluatorPreprocessor {
+  constructor(stream) {
+    super(stream);
+    this.stream = stream;
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    const result = {
+      fontSize: 0,
+      fontName: "",
+      fontColor: /* black = */ new Uint8ClampedArray(3),
+    };
+
+    let scaleFactor = 1;
```

Comment:
Sorry to complicate things further here, but shouldn't this be reset to a *previous* value on `OPS.restore`?

Reply:
That said it's strange to ship something which is know to be buggy... Let's fix the save/restore case.